eda/src/autogluon/eda/analysis/transform.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from typing import List, Optional
+
+import pandas as pd
+
+from autogluon.features import AbstractFeatureGenerator, AutoMLPipelineFeatureGenerator
+from .base import AbstractAnalysis
+from ..state import AnalysisState
+from ..state import StateCheckMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ApplyFeatureGenerator(AbstractAnalysis, StateCheckMixin):
+    """
+    This wrapper provides transformed features to all `children` shadowing outer datasets with the updated one after application of FeatureGenerator.
+
+    Parameters
+    ----------
+    category_to_numbers: bool, default = False
+        if `True', then transform `category` variables into their codes. This is useful when wrapped analyses expect numeric values
+    feature_generator: Optional[AbstractFeatureGenerator], default = None
+        feature generator to use for the transformation. If `None` is provided then `AutoMLPipelineFeatureGenerator` is applied.
+    parent: Optional[AbstractAnalysis], default = None
+        parent Analysis
+    children: List[AbstractAnalysis], default []
+        wrapped analyses; these will receive sampled `args` during `fit` call
+    kwargs
+
+    See also :func:`autogluon.features.AbstractFeatureGenerator`
+
+    Examples
+    --------
+    >>> from autogluon.eda.analysis.base import BaseAnalysis, Namespace
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> df_train = pd.DataFrame(...)
+    >>> df_test = pd.DataFrame(...)
+    >>>
+    >>> analysis = BaseAnalysis(train_data=df_train, test_data=df_test, label='D', children=[
+    >>>     Namespace(namespace='feature_generator_numbers', children=[
+    >>>         ApplyFeatureGenerator(category_to_numbers=True, children=[
+    >>>             # SomeAnalysis()  # This analysis will be called with transformed `train_data` and `test_data`
+    >>>         ])
+    >>>     ]),
+    >>>     # SomeAnalysis()  # This analysis will be called with the original features
+    >>> ])
+
+    """
+
+    def __init__(self,
+                 parent: Optional[AbstractAnalysis] = None,
+                 children: List[AbstractAnalysis] = [],
+                 state: Optional[AnalysisState] = None,
+                 category_to_numbers: bool = False,
+                 feature_generator: Optional[AbstractFeatureGenerator] = None,
+                 **kwargs) -> None:
+        super().__init__(parent, children, state, **kwargs)
+        self.category_to_numbers = category_to_numbers
+        if feature_generator is None:
+            feature_generator = AutoMLPipelineFeatureGenerator(
+                enable_numeric_features=True,
```

Comment:
Wondering if we can just use defaults instead of providing overrides which might change over time

Reply:
Feature generators are stateful, we don't want to initialize an object in defaults.